andy/src/main/java/nl/tudelft/cse1110/andy/execution/step/SourceCodeSecurityCheckStep.java

```diff
@@ -90,14 +92,32 @@ private boolean checkForKeywords(String code, ResultBuilder result) {
         );
         for (String keyword : keywords.keySet()) {
             if (code.contains(keyword)) {
-                result.compilationSecurityFail(keywords.get(keyword));
-                return false;
+                String[] lines = code.split("\\n");
+                // Split lines and search every line for keyword
+                for (int lineNumber = 0; lineNumber < lines.length; lineNumber++)
+                    if (lines[lineNumber].contains(keyword)) {
+                        // Adding 1 to lineNumber to convert from zero-based index to one-based index
+                        result.compilationSecurityFail(keywords.get(keyword), Optional.of(lineNumber+1));
+                        return false;
+                    }
             }
         }
 
         return true;
     }
 
+    public String removeComments(String code) {
```

Comment:
Should this be a `public` method, or `private`?

Reply:
Private 😓